### ISSUE TITLE
Button: Should brackets always be drawn at full width?

### ISSUE THREAD

**Issue by tig** [migueldeicaza/gui.cs#636]
@BDisp nicely added support for `Button` having width > their text, along with `TextAlignment` support.

Visually, I think it looks weird that the brackets (`[` and `]`) are not drawn all the way at full width:

![image](https://user-images.githubusercontent.com/585482/83955659-74c27480-a812-11ea-9e2a-500a91ea682e.png)

I was also considering changing the visual style of `Button` to use a better glyph than `[' to make them look better. Eg. `U+0254` as in  `┃Button Text┃`. 

Thoughts?

**Comment by BDisp** [comment]
I think that what currently exists is more framed in a `Button`. Only the horizontal lines are missing because it is already known that for that it would have a `FrameView `and we would be left with a button with three rows of height which would be excessively high.

**Comment by BDisp** [comment]
Unless we create a font where it may be possible to have underlined and overlined letters at the same time :-)

**Comment by tig** [comment]
TurboVision drew drop-shadows on buttons (assuming there was space available). It was a great look. I've been thinking about that too.

![image](https://user-images.githubusercontent.com/585482/83955990-1d260800-a816-11ea-91f5-17e36d6a780f.png)

We could support two styles:

1. Legacy gui.cs: `[ Button ]`
2. TurboVision: 

![image](https://user-images.githubusercontent.com/585482/83956010-68401b00-a816-11ea-9138-38602370a901.png)

I have a plan to support Bold and other ANSI formatting. Once that's done I was planning on making the drawing of the `<` and `>` in default buttons only happen if the ConsoleDriver didn't support bold (which will be possible).

**Comment by tig** [comment]
BTW, this code:

```
case TextAlignment.Centered:
	start = Frame.Width / 2 - caption.Length / 2;
	caption = $"{new string (' ', start)}{caption}{new string (' ', Frame.Width - caption.Length - start)}";
	if (c_hot_pos > -1) {
		c_hot_pos += start;
	}
	break;
```

Doesn't work right if `shown_text` includes unicode chars. You need to use ustring, and sadly, this means you can't use string interpolation...

**Comment by BDisp** [comment]
Thanks for the repair. I haven't tested it with unicode. I will take a look. If in the meantime you come up with a solution, please proceed.

**Comment by BDisp** [comment]
> TurboVision drew drop-shadows on buttons (assuming there was space available). It was a great look. I've been thinking about that too.

It really does look fantastic.

**Comment by tig** [comment]
![image](https://user-images.githubusercontent.com/585482/83956217-be15c280-a818-11ea-8e27-609fc8ffc23d.png)

**Comment by BDisp** [comment]
Rounded brackets seems fine.

**Comment by BDisp** [comment]
> 
> 
> BTW, this code:
> 
> ```
> case TextAlignment.Centered:
> 	start = Frame.Width / 2 - caption.Length / 2;
> 	caption = $"{new string (' ', start)}{caption}{new string (' ', Frame.Width - caption.Length - start)}";
> 	if (c_hot_pos > -1) {
> 		c_hot_pos += start;
> 	}
> 	break;
> ```
> 
> Doesn't work right if `shown_text` includes unicode chars. You need to use ustring, and sadly, this means you can't use string interpolation...

Fixed in PR #647

**Comment by tig** [comment]
I vote #647 nicely closes this.
